functions/zoom-downloader.py

```diff
@@ -38,24 +35,22 @@ def handler(event, context):
     setup_logging(context)
     logger.info(event)
 
-    if 'Records' not in event:
-        logger.error("No records in event. {}".format(event))
-        raise BadDynamoStreamEvent
+    download_queue = sqs.get_queue_by_name(QueueName=DOWNLOAD_QUEUE_NAME)
+    logger.debug("got queue {}".format(str(download_queue)))
 
-    if len(event['Records']) > 1:
-        logger.error("DynamoDB stream should be set to BatchSize: 1")
-        raise BadDynamoStreamEvent
-
-    event_type = event['Records'][0]['eventName']
-    logger.info("got event type {}".format(event_type))
-
-    if event_type != "INSERT":
+    try:
+        logger.debug("fetching a message...")
+        message = download_queue.receive_messages(MaxNumberOfMessages=1)[0]
+        logger.debug({'queue_message': message})
+    except IndexError:
+        logger.warning("No uploads ready for processing")
         return
 
-    record = json.loads(event['Records'][0]['dynamodb']['NewImage']['recording_data']['S'])
-    logger.debug(record)
+    download_data = json.loads(message.body)
+    logger.info(download_data)
+    recording_data = download_data['recording_data']
 
-    chronological_files = sorted(record['recording_files'], key=lambda k: k['recording_start'])
+    chronological_files = sorted(recording_data['recording_files'], key=lambda k: k['recording_start'])
```

Comment:
Another useful approach to this:
```
from operator import itemgetter
...
sorted(recording_data['recording_files'], key=itemgetter('recording_start'))
```
Not really better in this case, but in a situation where performance was a big deal.